app/src/main/java/org/sagebionetworks/research/mpower/tracking/fragment/RecyclerViewTrackingFragment.java

```diff
@@ -0,0 +1,118 @@
+package org.sagebionetworks.research.mpower.tracking.fragment;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+
+import android.graphics.drawable.Drawable;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.view.OnApplyWindowInsetsListener;
+import android.support.v4.view.ViewCompat;
+import android.support.v7.widget.DividerItemDecoration;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.support.v7.widget.RecyclerView.ItemDecoration;
+import android.support.v7.widget.RecyclerView.LayoutManager;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import org.sagebionetworks.research.mobile_ui.show_step.view.SystemWindowHelper;
+import org.sagebionetworks.research.mobile_ui.show_step.view.SystemWindowHelper.Direction;
+import org.sagebionetworks.research.mobile_ui.widget.ActionButton;
+import org.sagebionetworks.research.mobile_ui.widget.NavigationActionBar;
+import org.sagebionetworks.research.mpower.R;
+import org.sagebionetworks.research.mpower.tracking.view_model.TrackingTaskViewModel;
+import org.sagebionetworks.research.mpower.tracking.view_model.configs.TrackingItemConfig;
+import org.sagebionetworks.research.mpower.tracking.view_model.logs.TrackingItemLog;
+
+import butterknife.BindView;
+
+/**
+ * A RecyclerViewTrackingFragment allows for easier creation of TrackingFragments that use a recycler view to display
+ * their options. It requires a subclass to override initializeAdapter to give the recycler view it's adapter but handles
+ * the logic of setting up the recycler view, and laying it out appropriately.
+ * @param <ConfigType> The type of TrackingItemConfig.
+ * @param <LogType> The type of TrackingItemLog.
+ * @param <ViewModelType> The type of TrackingTaskViewModel.
+ */
+public abstract class RecyclerViewTrackingFragment
+        <ConfigType extends TrackingItemConfig, LogType extends TrackingItemLog, ViewModelType extends TrackingTaskViewModel<ConfigType, LogType>>
+        extends TrackingFragment<ConfigType, LogType, ViewModelType> {
+    @BindView(R.id.rs2_recycler_view)
```

Comment:
New code probably shouldn't depend on the 3rd party lib butterknife.
Kotlin has this functionality built in, but I'm pretty sure it just uses [Android Data Binding](https://developer.android.com/topic/libraries/data-binding/) which is available in Java.

Reply:
This should be a conversation with @liujoshua, All of SageResearch mobile-ui has a dependency on ButterKnife as well so this isn't introducing the dependency into the project.